### DISCUSSION TITLE
Raw query string?

### DISCUSSION THREAD

**Discussion by sbking** [leptos-rs/leptos#1545] [Q&A] [answered]
I'm trying to access the raw query string so I can parse it with `serde_qs` instead of `use_query`. I need to do this so that I can return form field values and errors in the query string to support progressive enhancement for forms when JS/WASM is not enabled (see #1544).

However I can't find any way with `leptos_router` to access the raw query string. I can get a `ParamsMap` through `use_location`, but when I call `to_query_string` on it, the keys all get url-escaped and I can no longer parse it with `serde_qs`.

**Comment by gbj** [comment, marked as answer]
The `search` field of the `Location` that you get from `use_location`.

https://docs.rs/leptos_router/latest/leptos_router/struct.Location.html

**Reply by sbking** [reply]
d'oh, thanks, tunnel vision!
